Tensile/Tests/unit/test_TensileCreateLibrary.py

```diff
@@ -352,9 +353,44 @@ def test_generateMasterFileList():
     assert isinstance(t[1], MasterLibraryMock), "Incorrect type for value."
     assert t[1].data == (idx+2), "Incorrect data."        
 
+
+def test_componentsOfLogicDataAndSolutionsConstruction(initGlobalParametersForTCL):
+    
+    requiredArgs = ["--jobs=2", "/unused/logic/path", "/unused/output/path", "HIP"]
+    
+    with initGlobalParametersForTCL(["--architecture=gfx900"]+requiredArgs):
+        for separateArch in [False, True]:
+            rootPath = Path(__file__).parent/"../test_data"/"unit"/"solutions"
+            yamlFiles = [rootPath/f for f in["vega10_Cijk_Ailk_Bjlk_CB_GB.yaml", "hip_Cijk_Ailk_Bjlk_CB_GB.yaml"]]
```

Comment:
I've implemented similar logic for reading a yaml file in another PR. I opted to use the pytest [request](https://docs.pytest.org/en/latest/reference/reference.html#request) fixture, provides the same end result. We should chat about how we want to maintain testing resource files such as these in a consistent manner.

Reply:
Reconcile in when we merge two different approaches?